### PR TITLE
Fix footnote rendering issues

### PR DESCRIPTION
**Pull request by mruwnik**

### CODE REVIEW

File: app/routes/index.tsx [hunk 1]
```diff
@@ -113,13 +120,14 @@ export default function App() {
...
-      el.classList.toggle('visible')
-      const container = el.closest('.react-auto-height') as HTMLElement
-      container.style.removeProperty('height')
-
+      showMore(el)
       e.preventDefault()
       return
     }
+    if (el.parentElement?.classList.contains('footnote-ref')) {
+      showMore(el)
```
[the remark this answers is not in the document]
Comment (Aprillion) [reply app/routes/index.tsx:128]: @jrhender you know how we thought tests around server utils are on top of priorities for testing?

I am starting to feel that end-to-end testing of the rendering of markdown that comes out of parsing Google Docs might be equally important in near future 😓 no need to change your priorities, just to keep in mind that just unit tests for server utils might not be enough, there are just so many moving parts all the way from Google Docs to browser DOM that we are fixing more and more edge cases over time and sooner or later we will want to test those so a new fix won't brake an old fix 😿

[the remark this answers is not in the document]
Comment (mruwnik) [reply app/routes/index.tsx:128]: yes! This! Please!

[the remark this answers is not in the document]
Comment (jrhender) [reply app/routes/index.tsx:128]: @Aprillion thanks for the heads up. I think even tests that assert again the DOM would benefit from scaffolding that would be required for server-utils tests. By this I mean that the calls to coda & cloudflare would still be mocked as a part of these higher level tests that you envision, right?
